src/python/zquantum/qaoa/maxcut.py

```diff
@@ -2,37 +2,103 @@
 import numpy as np
 from itertools import combinations
 from random import uniform
-
-
 from openfermion import QubitOperator
 from zquantum.core.utils import dec2bin
-from zquantum.core.graph import generate_graph_node_dict
+from zquantum.core.graph import generate_graph_node_dict, generate_graph_from_specs
+from typing import Dict, List, Union
+
+from .farhi_ansatz import QAOAFarhiAnsatz
+
+
+def create_farhi_qaoa_circuits(
+    hamiltonians: List[QubitOperator], number_of_layers: int
+):
+    """Creates parameterizable quantum circuits based on the farhi qaoa ansatz for each 
+    hamiltonian in the input list using the set number of layers. 
+
+    Args:
+        hamiltonians (List[QubitOperator]): List of hamiltonians for constructing the
+            circuits
+        number_of_layers (int): The number of layers of the ansatz in the circuit
+                
+    Returns:
+        List of zquantum.core.circuit.Circuit
+    """
+    circuitset = []
+    for hamiltonian in hamiltonians:
+        ansatz = QAOAFarhiAnsatz(number_of_layers, hamiltonian)
+        circuitset.append(ansatz.parametrized_circuit)
+    return circuitset
+
+
+def get_random_maxcut_hamiltonians(
+    graph_specs: Dict,
+    number_of_instances: int,
+    number_of_qubits: Union[int, List[int]],
+    **kwargs
+):
+    """Generates random maxcut hamiltonians based on the input graph description for a range 
+    of number of qubits and a set number of instances.
+
+    Args:
+        graph_specs (dict): Specifications of the graph to generate. It should contain at 
+            least an entry with key 'type_graph' (Note: 'num_nodes' key will be overwritten)
+        number_of_instances (int): The number of hamiltonians to generate
+        number_of_qubits (int or List[int]): The number of qubits in the hamiltonian. If this is a list, then
+            a random value will be picked to generate each instance.
```

Comment:
I assume you mean "random value from the list"?

Reply:
Yes: changed in commit 439a707902cc4fcae3fcffdaa112793e59dfe1f6